edx_proctoring/static/proctoring/js/views/proctored_exam_view.js

```diff
@@ -78,6 +78,10 @@ var edx = edx || {};
                 // remove callback on unload event
                 $(window).unbind('beforeunload', this.unloadMessage);
             }
+            var desktopApplicationJsUrl = this.model.get('desktop_application_js_url');
+            if (desktopApplicationJsUrl && !edx.courseware.proctored_exam.configuredWorkerURL) {
```

Comment:
For starting and ending the exam, the global variable `configuredWorkerURL` is set directly before `examStartHandler` and `examEndHandler` (which your link points to) are attached as listeners to DOM events. That setting happens in django templates that are specific to the start- and end-specific pages of the proctoring flow.

For the purposes of ping, we can't apply a similar approach, since ping needs to happen all over courseware. That's why we set it here as well. In the special case of the ready_to_start and ready_to_submit pages, we'll see `edx.courseware.proctored_exam.configuredWorkerURL` already set and not try to set it again.